lib/newrelic-roda/instrument.rb

```diff
@@ -23,20 +23,35 @@ def match_all(args)
           end
           add_method_tracer :match_all
 
+          def filter_params(param_hash)         
+            filtered_params = ENV['FILTERED_PARAMS']&.split(',')
+            param_hash.with_indifferent_access.except(*filtered_params)
+          end
+
           def build_params
```

Comment:
If we call params - we start decrypting the thing - also - parsing software is one of the main attack vectors on a system - so I'd be super-careful here and make things super-clear and least ambiguous in the code,

### Plan A

I'd try to figure out what the input is - ie:

1. if the body, after whitespace, starts with a `{` - then assume JSON - and parse it
2. else - let's cross that bridge when we get here - let's see what these scenarios are - and if we can cook something using HTTP headers at this point

To my knowledge we have POST & GET coming via the QR code payment routes & the file upload - so we'll need to capture those somehow

### Plan B

Extend `NewRelic::Agent::Instrumentation::Roda` to allow us to set a `build_params` lambda that receives the URI & request object and then we can have an if-else in infoswitch that takes into account which path, which params etc etc

This may be the more robust version actually